src/repl/mod.rs

```diff
@@ -177,7 +177,7 @@ impl ReplImpl {
         match term {
             ExtendedTerm::RichTerm(t) => {
                 let t = prepare(self, None, t)?;
-                Ok(eval_function(t, &self.env.eval_env, &mut self.cache)?.into())
+                Ok(eval_function(&mut eval::VirtualMachine::new(&mut self.cache), t, &self.env.eval_env)?.into())
```

Comment:
That's not very important, but in the REPL, we are evaluating things repeatedly. I wonder if that wouldn't be a tad cleaner to store a `vm` in the state of the REPL and to reuse it for each evaluation, with a prior call to `reset()`

Reply:
Sounds good (small PRs are always good :slightly_smiling_face: )